pkg/standalone-utils/test/LidoRelayer.test.ts

```diff
@@ -1,462 +1,950 @@
+          ).wait();
 
-        await vault.authorizer
-          ?.connect(admin)
-          .grantRolesGlobally([manageUserBalanceAction, joinAction], relayer.address);
-      });
+          const relayerIsRecipient = TypesConverter.toAddress(tokenRecipient) === relayer.address;
+          expectTransferEvent(
+            receipt,
+            {
+              from: relayerIsRecipient ? ZERO_ADDRESS : relayer.address,
+              to: relayerIsRecipient ? relayer.address : TypesConverter.toAddress(tokenRecipient),
+              value: expectedWstETHAmount,
+            },
+            wstETH
+          );
+        });
+
+        it('stores stake output as chained reference', async () => {
+          const amount = fp(1);
+          const expectedWstETHAmount = await wstETH.instance.getWstETHByStETH(amount);
+
+          const receipt = await (
+            await relayer
+              .connect(sender)
+              .multicall([encodeStakeETHAndWrap(tokenRecipient, amount, toChainedReference(0))], { value: amount })
+          ).wait();
+
+          const relayerIsRecipient = TypesConverter.toAddress(tokenRecipient) === relayer.address;
+          expectTransferEvent(
+            receipt,
+            {
+              from: relayerIsRecipient ? ZERO_ADDRESS : relayer.address,
+              to: relayerIsRecipient ? relayer.address : TypesConverter.toAddress(tokenRecipient),
+              value: expectedWstETHAmount,
+            },
+            wstETH
+          );
 
-      context('when the user did allow the relayer', () => {
-        sharedBeforeEach('allow relayer', async () => {
-          await vault.instance.connect(sender).setRelayerApproval(sender.address, relayer.address, true);
+          await expectChainedReferenceContents(toChainedReference(0), expectedWstETHAmount);
         });
 
-        it('joins the pool', async () => {
-          const receipt = await relayer
-            .connect(sender)
-            .joinPool(basePoolId, sender.address, sender.address, joinRequest);
+        it('stakes with chained references', async () => {
+          const amount = fp(1);
+          const expectedWstETHAmount = await wstETH.instance.getWstETHByStETH(amount);
 
-          expectEvent.inIndirectReceipt(await receipt.wait(), vault.instance.interface, 'PoolBalanceChanged', {
-            poolId: basePoolId,
-            liquidityProvider: sender.address,
-          });
+          await setChainedReferenceContents(toChainedReference(0), amount);
+
+          const receipt = await (
+            await relayer
+              .connect(sender)
+              .multicall([encodeStakeETHAndWrap(tokenRecipient, toChainedReference(0))], { value: amount })
+          ).wait();
+
+          expectTransferEvent(
+            receipt,
+            { from: ZERO_ADDRESS, to: relayer.address, value: expectedWstETHAmount },
+            wstETH
```

Comment:
Same re. we should test that the input was the chained reference instead of indirectly asserting that via the output